opentelemetry-api/src/opentelemetry/distributedcontext/__init__.py

```diff
@@ -11,3 +11,57 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from contextlib import contextmanager
+import string
+import typing
+
+
+class EntryMetadata:
+    NO_PROPAGATION = 0
+    UNLIMITED_PROPAGATION = -1
+
+    def __init__(self, entry_ttl: int) -> None:
+        self.entry_ttl = entry_ttl
+
+
+class EntryKey(str):
+    def __new__(cls, value):
+        if any(c not in string.printable for c in value) or len(value) > 255:
+            raise ValueError("Invalid EntryKey", value)
+        return str.__new__(cls, value)
+
+
+class EntryValue(str):
+    def __new__(cls, value):
+        if any(c not in string.printable for c in value):
+            raise ValueError("Invalid EntryValue", value)
+        return str.__new__(cls, value)
+
+
+class Entry:
+    def __init__(
+        self, metadata: EntryMetadata, key: EntryKey, value: EntryValue
+    ) -> None:
+        self.metadata = metadata
+        self.key = key
+        self.value = value
+
+
+class DistributedContext:
+    def get_entries(self) -> typing.Iterable[Entry]:
+        pass
```

Comment:
We've been using ABC's to define the interfaces. this has a few advantages, including ensuring that unimplemented APIs are caught on startup (including unit tests) rather than when a live application tries to execute a function for the first time.